src/Explorer/Explorer.tsx

```diff
@@ -1394,7 +1394,11 @@ export default class Explorer {
       tab.hashLocation().startsWith(hashLocation)
     ) as TerminalTab[];
 
-    const index = terminalTabs.length + 1;
+    let index = 1;
+    if (terminalTabs.length > 0) {
+      index = terminalTabs[terminalTabs.length - 1].getIndex() + 1;
```

Comment:
What happens if I open 3 tabs, close tab1 and open a new one? Looks like it'll name it tab3 with this logic when we already have one with the same name?